### PR TITLE
i#2985 drx_expand_scatter_gather(): Add all mask restore related AVX2 test variants.

### PR DESCRIPTION
**Pull request by hgreving2304**
Adds an update destination mask test to client.drx-scattergather for AVX2. As with
previous tests of this kind, the error is currently expected until a drx restore event
has been implemented.

Does some renaming into AVX2/AVX-512 function names.

Initializes the pc variables with INT_MAX in order to add robustness to the test.

Issue: #2985

### CODE REVIEW

File: suite/tests/client-interface/drx-scattergather-shared.h [hunk 1]
```diff
@@ -30,7 +30,8 @@
  * DAMAGE.
```
Comment (derekbruening) [inline suite/tests/client-interface/drx-scattergather-shared.h]: > Issue: 2985

Missing `#`.

File: suite/tests/client-interface/drx-scattergather.c [hunk 2]
```diff
@@ -235,6 +241,20 @@ signal_handler_check_k1(int sig, siginfo_t *siginfo, ucontext_t *ucxt)
...
 }
+
+static void
+signal_handler_check_ymm1(int sig, siginfo_t *siginfo, ucontext_t *ucxt)
+{
+    kernel_fpstate_t *fp = (kernel_fpstate_t *)ucxt->uc_mcontext.fpregs;
+#        ifdef X64
+    if (((fp->xmm_space[8] >> 31) & 0x1) != 0) {
+#        else
+    if (((fp->_xmm[2].element[3] >> 31) & 0x1) != 0) {
```
Comment (derekbruening) [inline suite/tests/client-interface/drx-scattergather.c]: Please add either a comment pointing out where the code is that writes to this particular precise register part (and probably a comment by that code too about this linked code here), or use named constants if that's straightforward.

Ditto for the xmm_space[8]'s 32nd bit above.

File: suite/tests/client-interface/drx-scattergather-shared.h (hunk 1 above)
Comment (hgreving2304) [reply suite/tests/client-interface/drx-scattergather-shared.h]: Fixed.

File: suite/tests/client-interface/drx-scattergather.c (hunk 2 above)
Comment (hgreving2304) [reply suite/tests/client-interface/drx-scattergather.c]: Done.

**Pull request merged**